### ISSUE TITLE
[Bug]: Notification filters should use []FilterExpressions type

### ISSUE THREAD

**Issue by hgiasac** [OneSignal/onesignal-go-api#24]
### What's on your mind?

Currently, we are impossible to input filters with OR condition into the `Notification` model.

```json
"filters": [
  {"field": "tag", "key": "level", "relation": "=", "value": "10"}, 
  {"operator": "OR"}, {"field": "tag", "key": "level", "relation": "=", "value": "20"}
]
```

The type of the [`Filters`](https://github.com/OneSignal/onesignal-go-api/blob/c9a3cc74fcc54c45833706a55b427437f1b70f1a/model_notification.go#L214) field should be `[]FilterExpressions`.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by sherwinski** [comment]
Hello! Apologies that this question has gone stale.
If this question is still a current concern, please ping me and we will investigate.

**Comment by shebinleo** [comment]
@sherwinski  Yes, it is. Currently, there is no way to set the operator for the filter with an OR condition in the Notification model.

**Comment by sherwinski** [comment]
@shebinleo Noted, I will confer with the team to try and get this patched in a future release.

**Comment by dreamlabsg** [comment]
For now, I was able to do a workaround provided here https://github.com/OneSignal/onesignal-go-api/issues/27#issuecomment-1496652820

**Comment by sherwinski** [comment]
This has been fixed in the latest major release (https://github.com/OneSignal/onesignal-go-api/releases/tag/v5.2.0-beta1) so for anyone still running into this, I would definitely encourage upgrading library versions. Please feel free to comment here or open a new issue if there are still any questions about this.

Thanks
